### PR TITLE
Add interfaces for `StakingCredential`

### PR DESCRIPTION
**Pull request by SeungheonOh**
Now, one can use `address` to specify UTXO address with `StakingCredential` or instead one can use `withStakingCredential` to monoidally append `StakingCredential`.

### CODE REVIEW

**Review by kozross — CHANGES_REQUESTED** [review]
Version bump, `since`s, etc.

File: src/Plutarch/Context/Base.hs [hunk 1]
```diff
@@ -167,41 +185,100 @@ utxoToTxOut ::
...
 withRefIndex tidx = mempty{utxoTxIdx = Just tidx}
 
+{- | Specify `TxOutRef` of a UTXO.
+
+ @since 2.0.0
+-}
 withRef :: TxOutRef -> UTXO
 withRef (TxOutRef tid idx) = withRefTxId tid <> withRefIndex idx
 
+{- | Specify value of a UTXO. This will me monoidally merged values
```
Comment (peter-mlabs) [inline src/Plutarch/Context/Base.hs]: 'be', not 'me'.

**Review by peter-mlabs — APPROVED** [review]
Fixed a typo. We can merge once CI passes

File: src/Plutarch/Context/Base.hs (hunk 1 above)
Comment (SeungheonOh) [reply src/Plutarch/Context/Base.hs]: Thank you!

**Review by kozross — APPROVED** [review]

**Pull request merged**